packages/recommend-react/src/RecommendContext.tsx

```diff
@@ -0,0 +1,207 @@
+import { RecommendationsQuery, RecommendClient } from '@algolia/recommend';
+import { BatchKeyPair } from '@algolia/recommend-core';
+import {
+  BatchRecommendations,
+  getBatchRecommendations,
+} from '@algolia/recommend-core/src';
+import { dequal } from 'dequal';
+import React from 'react';
+
+import { isPresent } from './utils/isPresent';
+
+type GetParametersResult = {
+  queries: RecommendationsQuery[];
+  keyPair: BatchKeyPair;
+};
+
+type RecommendWidget<TObject> = {
+  getParameters: () => GetParametersResult;
+  onResult: (value: BatchRecommendations<TObject>) => void;
+  onRequest: () => void;
+  key: string;
+  param: GetParametersResult;
+};
+
+type RecommendContextType<TObject> = {
+  register: (
+    widget: Omit<RecommendWidget<TObject>, 'param'>
+  ) => (key: string) => void;
+  hasProvider: boolean;
```

Comment:
we can but it would make destructuring in the components a 2 lines operation instead of the straightforward way we do it today. 

I don't see the benefit of null in this case?